sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExprUtils.scala

```diff
@@ -168,7 +168,13 @@ object ExprUtils extends QueryErrorsBase {
         a.failAnalysis(
           errorClass = "MISSING_GROUP_BY",
           messageParameters = Map.empty)
-      case e: Attribute if !a.groupingExpressions.exists(_.semanticEquals(e)) =>
+      case e: Attribute if !a.groupingExpressions.exists(_.semanticEquals(e)) &&
+        !a.groupingExpressions.exists {
```

Comment:
why do we need this change?

Reply:
without it, it hits the `QueryCompilationErrors.columnNotInGroupByClauseError(e)`

with query plan:
```
Aggregate [collationkey(name#17)], [any_value(name#17, false) AS name#24, count(1) AS count(1)#20L]
+- LogicalRDD [name#17], false
```

so name#17 is not found in groupingExpressions